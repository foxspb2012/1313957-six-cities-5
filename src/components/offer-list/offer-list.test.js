import React from 'react';
import {Router} from 'react-router-dom';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import OfferList from './offer-list';
import history from '../../history';

const NameSpace = {
  APP: `APP`,
  USER: `USER`,
  OFFERS: `OFFERS`,
  REVIEWS: `REVIEWS`,
};

const getOffersMock = (count) => {
  const templateOffers = Array(count)
    .fill(``)
    .map((_, index) => {
      return {
        'id': index + 1,
        'city': {
          'name': `Paris`,
          'location': {
            'coordinates': [48.85661, 2.351499],
            'zoom': 13
          }
        },
        'type': `room`,
        'name': `Wood and stone place`,
        'description': `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
        'photos': [`img/apartment-01.jpg`, `img/apartment-03.jpg`],
        'isFavorite': Boolean(Math.random()),
        'isPremium': Boolean(Math.random()),
        'rating': 1,
        'price': 100,
        'bedroomAmount': 1,
        'guestAmount': 1,
        'features': [`Dishwasher`, `Baby seat`],
        'host': {
          'name': `Host`,
          'avatar': `img/avatar-max.jpg`,
          'isPro': false,
          'id': index + 1
        },
        'location': {
          'coordinates': [48.87961000000001, 2.353499],
          'zoom': 16
        },
        'mainPhoto': `img/apartment-03.jpg`,
      };
    });

  if (count === 1) {
    const [oneOffer] = templateOffers;
    return oneOffer;
  }

  return templateOffers;
};

const mockState = {
  [NameSpace.APP]: {
    activeCity: `Paris`,
    activeSortType: `popular`,
    activeOffer: null
  },
  [NameSpace.OFFERS]: {
    offers: [],
    offer: {},
    nearOffers: [],
    favoriteOffers: [],
    error: null,
    isLoaded: false,
    isLoadedOffer: false,
    isLoadedNearOffers: false,
    isLoadedFavoritesOffers: false,
  },
  [NameSpace.REVIEWS]: {
    reviews: [],
    isReviewSending: false,
    error: null
  },
  [NameSpace.USER]: {
    authorizationStatus: `UNAUTHORIZED`,
    email: null
  }
};
const OffersListType = {
  MAIN: `MAIN`,
  NEAR: `NEAR`,
  FAVORITES: `FAVORITES`,
};

describe(`Should OfferList render correctly`, () => {
  it(`Should OfferList render correctly empty component`, () => {
    const mockStore = configureStore();
    const store = mockStore(mockState);
    const tree = renderer
      .create(
          <Provider store={store}>
            <Router history={history}>
              <OfferList
                typeClass={OffersListType.FAVORITES}
              />
            </Router>
          </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should OfferList render correctly with offers and favorites type`, () => {
    const mockStore = configureStore();
    const store = mockStore(mockState);
    const tree = renderer
      .create(
          <Provider store={store}>
            <Router history={history}>
              <OfferList
                typeClass={OffersListType.FAVORITES}
                offer={getOffersMock(5)}
              />
            </Router>
          </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should OfferList render correctly with offers and main type`, () => {
    const mockStore = configureStore();
    const store = mockStore(mockState);
    const tree = renderer
      .create(
          <Provider store={store}>
            <Router history={history}>
              <OfferList
                typeClass={OffersListType.MAIN}
                offer={getOffersMock(5)}
              />
            </Router>
          </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should OfferList render correctly with offers and near type`, () => {
    const mockStore = configureStore();
    const store = mockStore(mockState);
    const tree = renderer
      .create(
          <Provider store={store}>
            <Router history={history}>
              <OfferList
                typeClass={OffersListType.NEAR}
                offer={getOffersMock(5)}
              />
            </Router>
          </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
