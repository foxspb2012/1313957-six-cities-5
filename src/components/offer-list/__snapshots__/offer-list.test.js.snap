// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should OfferList render correctly Should OfferList render correctly empty component 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
      "fontSize": "38px",
      "fontStyle": "oblique",
      "fontWeight": 700,
      "justifyContent": "flex-start",
      "lineHeight": 0.7,
      "margin": "20px",
      "padding": "30px",
      "textAlign": "center",
      "width": "100%",
    }
  }
>
  <h3
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "space-between",
      }
    }
  >
    <img
      alt="logo"
      height="50"
      src="/img/logo.svg"
      style={
        Object {
          "marginRight": "20px",
        }
      }
      width="100"
    />
    <svg
      height="100%"
      version="1.1"
      width="100%"
      xmlns="http://www.w3.org/2000/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <path
        id="path"
      >
        <animate
          attributeName="d"
          begin="0s"
          dur="3s"
          from="m0,110 h0"
          repeatCount="indefinite"
          to="m0,110 h1100"
        />
      </path>
      <text
        fill="black"
      >
        <textPath
          xlinkHref="#path"
        >
          Loading...
        </textPath>
      </text>
    </svg>
  </h3>
</div>
`;

exports[`Should OfferList render correctly Should OfferList render correctly with offers and favorites type 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
      "fontSize": "38px",
      "fontStyle": "oblique",
      "fontWeight": 700,
      "justifyContent": "flex-start",
      "lineHeight": 0.7,
      "margin": "20px",
      "padding": "30px",
      "textAlign": "center",
      "width": "100%",
    }
  }
>
  <h3
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "space-between",
      }
    }
  >
    <img
      alt="logo"
      height="50"
      src="/img/logo.svg"
      style={
        Object {
          "marginRight": "20px",
        }
      }
      width="100"
    />
    <svg
      height="100%"
      version="1.1"
      width="100%"
      xmlns="http://www.w3.org/2000/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <path
        id="path"
      >
        <animate
          attributeName="d"
          begin="0s"
          dur="3s"
          from="m0,110 h0"
          repeatCount="indefinite"
          to="m0,110 h1100"
        />
      </path>
      <text
        fill="black"
      >
        <textPath
          xlinkHref="#path"
        >
          Loading...
        </textPath>
      </text>
    </svg>
  </h3>
</div>
`;

exports[`Should OfferList render correctly Should OfferList render correctly with offers and main type 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
      "fontSize": "38px",
      "fontStyle": "oblique",
      "fontWeight": 700,
      "justifyContent": "flex-start",
      "lineHeight": 0.7,
      "margin": "20px",
      "padding": "30px",
      "textAlign": "center",
      "width": "100%",
    }
  }
>
  <h3
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "space-between",
      }
    }
  >
    <img
      alt="logo"
      height="50"
      src="/img/logo.svg"
      style={
        Object {
          "marginRight": "20px",
        }
      }
      width="100"
    />
    <svg
      height="100%"
      version="1.1"
      width="100%"
      xmlns="http://www.w3.org/2000/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <path
        id="path"
      >
        <animate
          attributeName="d"
          begin="0s"
          dur="3s"
          from="m0,110 h0"
          repeatCount="indefinite"
          to="m0,110 h1100"
        />
      </path>
      <text
        fill="black"
      >
        <textPath
          xlinkHref="#path"
        >
          Loading...
        </textPath>
      </text>
    </svg>
  </h3>
</div>
`;

exports[`Should OfferList render correctly Should OfferList render correctly with offers and near type 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
      "fontSize": "38px",
      "fontStyle": "oblique",
      "fontWeight": 700,
      "justifyContent": "flex-start",
      "lineHeight": 0.7,
      "margin": "20px",
      "padding": "30px",
      "textAlign": "center",
      "width": "100%",
    }
  }
>
  <h3
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "space-between",
      }
    }
  >
    <img
      alt="logo"
      height="50"
      src="/img/logo.svg"
      style={
        Object {
          "marginRight": "20px",
        }
      }
      width="100"
    />
    <svg
      height="100%"
      version="1.1"
      width="100%"
      xmlns="http://www.w3.org/2000/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <path
        id="path"
      >
        <animate
          attributeName="d"
          begin="0s"
          dur="3s"
          from="m0,110 h0"
          repeatCount="indefinite"
          to="m0,110 h1100"
        />
      </path>
      <text
        fill="black"
      >
        <textPath
          xlinkHref="#path"
        >
          Loading...
        </textPath>
      </text>
    </svg>
  </h3>
</div>
`;
